Release 1.12.0 (#5014)
antrea-io/antrea: CHANGELOG/CHANGELOG-1.12.md

# Changelog 1.12

## 1.12.0 - 2023-05-24

- The Multicast, TopologyAwareHints, NodeIPAM features are graduated from Alpha to Beta and therefore enabled by default.

### Added

- Add a new field 'logLabel' to ACNP and ANP CRDs to specify firewall log label for Audit Logging. ([#4748](https://github.com/antrea-io/antrea/pull/4748), [@qiyueyao])

> - Add support for short-circuiting in AntreaProxy to ensure that the traffic from Pod/Node clients to

Author: Moved to `Changed`

external addresses behaves the same way as the traffic from external clients to external addresses. ([#4815](https://github.com/antrea-io/antrea/pull/4815), [@hongliangl])
- Add support for ExternalIP in AntreaProxy to allows a Service to be accessed from outside the cluster using a static IP address. ([#4866](https://github.com/antrea-io/antrea/pull/4866), [@hongliangl])
- Add two new fields 'sourcePort' and 'sourceEndPort' in Antrea-native policy API to match traffic initiated from specific ports. ([#4687](https://github.com/antrea-io/antrea/pull/4687), [@Dyanngg])
- Add OVS table name as label for 'ovs_flow_count' Prometheus metrics. ([#4893](https://github.com/antrea-io/antrea/pull/4893), [@cr7258])
- Add WireGuard tunnel mode for Multi-cluster to establish encrypted connection between member clusters. ([#4606](https://github.com/antrea-io/antrea/pull/4606), [@hjiajing])

> - Watch EndpointSlice API by default for exported Service's endpoint change in Multi-cluster. ([#4895](https://github.com/antrea-io/antrea/pull/4895), [@luolanzone])

Author: I feel there is no need to highlight PodIP Endpoints at first sentence. MC also watches EndpointSlice API to detect if there is any available Endpoints for exported Service when it's ClusterIP type, and may remove the corresponding ResourceExport if there is no available Endpoints. Maybe we can say:

"Add support for EndpointSlice API for Multi-cluster Services. When the EndpointSlice API is available for the cluster, EndpointSlice resources of the exported Service, rather than the Endpoints resource, will be watched. For the Pod IP type Endpoints, EndpointSlice resources will be exported as the Multi-cluster Service's endpoints."

> - Watch EndpointSlice API by default for exported Service's endpoint change in Multi-cluster. ([#4895](https://github.com/antrea-io/antrea/pull/4895), [@luolanzone])

Reviewer: The thing is "watch" is more internal implementation and not very interesting to users. How about: "watched" -> "processed"

Then maybe remove "For the Pod IP type of Services, EndpointSlice resources will be exported as the Multi-cluster Service's endpoints." First, there is no "Pod IP type of Services" and no one will understand what does it mean. Second, finally we still export and create Endpoints, not EndpointSlice, and the sentence may be misleading to say "EndpointSlice resources will be exported".

> - Watch EndpointSlice API by default for exported Service's endpoint change in Multi-cluster. ([#4895](https://github.com/antrea-io/antrea/pull/4895), [@luolanzone])

Author: Updated.

### Changed

> - Deprecate Antrea Octant Plugin. ([#4825](https://github.com/antrea-io/antrea/pull/4825), [@antoninbas])

Reviewer: Question - we do not have any new UI enhancements in this release?

Author: UI is in another repo, I suppose there are some changes. Should we add UI PRs into this release log?
@antoninbas any suggestion for the UI change?

Reviewer 2: There is no need to mention UI changes as they are tracked separately.
However, you could mention the UI here:

> Deprecate Antrea Octant Plugin; it is replaced by a dedicated Antrea UI.

And also it could be mentioned in the release announcement for Antrea v1.12

- Update Open vSwitch to 2.17.6. ([#4959](https://github.com/antrea-io/antrea/pull/4959), [@tnqn])
- Update Windows OVS version to 2.16.7. ([#4705](https://github.com/antrea-io/antrea/pull/4705), [@XinShuYang])
- Stop serving v1alpha2 version of the ClusterGroup CRD. ([#4812](https://github.com/antrea-io/antrea/pull/4812), [@antoninbas])

> - Change to cache Flow modification message in Antrea Agent to improve Agent memory usage. ([#4495](https://github.com/antrea-io/antrea/pull/4495), [@wenyingd])

Reviewer: Flow -> flow

message -> messages

- Replace PacketIn/Controller with PacketIn2/Controller2 to improve packetin handler. ([#4768](https://github.com/antrea-io/antrea/pull/4768), [@GraysonWu])

> - Change to look up Pods by name instead of IP address to fetch labels in Flow Aggregator, to avoid obtaining incorrect Pods when Pod turnover quickly. ([#4942](https://github.com/antrea-io/antrea/pull/4942), [@dreamtalen])

Reviewer: when Pods turn over quickly

### Fixed

> 

Author: added some missing PRs. Thanks.

- Use LOCAL instead of CONTROLLER as the in_port of packet-out message to fix Windows agent crash issue. ([#4992](https://github.com/antrea-io/antrea/pull/4992), [@tnqn])
- Fix the Antrea Agent crash issue on Windows by running modules that rely on Services after AntreaProxy is ready. ([#4946](https://github.com/antrea-io/antrea/pull/4946), [@tnqn])

> - Fix a bug that a deleted policy is still enforced when a new policy with the same name exists. ([#4986](https://github.com/antrea-io/antrea/pull/4986), [@tnqn])

Reviewer: policy -> NetworkPolicy

> - Make FQDN NetworkPolicy work for upper case DNS. ([#4934](https://github.com/antrea-io/antrea/pull/4934), [@GraysonWu])

Reviewer: DNS -> FQDNs?

> - Fix a bug that K8s Networkpolicy audit logging not working for Service access. ([#4780](https://github.com/antrea-io/antrea/pull/4780), [@qiyueyao])

Reviewer 3: ```suggestion
- Fix a bug that K8s Networkpolicy audit logging doesn't work for Service access. ([#4780](https://github.com/antrea-io/antrea/pull/4780), [@qiyueyao])
```

- Fix Service not being updated correctly when stickyMaxAgeSeconds or InternalTrafficPolicy is updated. ([#4845](https://github.com/antrea-io/antrea/pull/4845), [@tnqn])
- Fix EndpointSlice API availablility check to resolve the issue that AntreaProxy always falls back to the Endpoints API when EndpointSlice is enabled. ([#4852](https://github.com/antrea-io/antrea/pull/4852), [@tnqn])

> - Prevent headless Service from updating Service CIDR to avoid overwriting the default route of host network unexpectedly. ([#5008](https://github.com/antrea-io/antrea/pull/5008), [@hongliangl])

Reviewer: Probably add "In Antrea Agent Service CIDR discovery," to be clear.

headless Service -> headless Services

the discovered Service CIDR

> - Use a random port when the UDP source port in a traceflow is 0. ([#4963](https://github.com/antrea-io/antrea/pull/4963), [@gran-vmv])

Reviewer: traceflow -> Traceflow

# Changelog 1.12

## 1.12.0 - 2023-05-24

> - The Multicast, TopologyAwareHints, NodeIPAM features are graduated from Alpha to Beta and therefore enabled by default.

Reviewer 4: We should mention the new `multicast.enable` configuration parameter.

Author: Added.

### Added

> - Add a new 'Failed' phase in ANP status for the case when all Agents have reported the status and at least one failure is received. ([#4608](https://github.com/antrea-io/antrea/pull/4608), [@wenyingd])

Reviewer: I feel this can be "Changed" too.

Author: Moved.

- Add two new fields 'sourcePort' and 'sourceEndPort' in Antrea-native policy API to match traffic initiated from specific ports. ([#4687](https://github.com/antrea-io/antrea/pull/4687), [@Dyanngg])
- Add a new field 'logLabel' to ACNP and ANP CRDs to specify firewall log label for Audit Logging. ([#4748](https://github.com/antrea-io/antrea/pull/4748), [@qiyueyao])

> - Add a new Controller configuration 'clientCAFile' to allow user to specify client CA. ([#4664](https://github.com/antrea-io/antrea/pull/4664), [@wenyingd])

Reviewer: Controller -> Antrea Controller

- Add support for ExternalIP in AntreaProxy to allow a Service to be accessed from outside the cluster using an external IP address. ([#4866](https://github.com/antrea-io/antrea/pull/4866), [@hongliangl])

> - Add OVS table name as label for 'ovs_flow_count' Prometheus metrics. ([#4893](https://github.com/antrea-io/antrea/pull/4893), [@cr7258])

Author: Moved to `Changed`

- Add WireGuard tunnel mode for Antrea Multi-cluster to support encryption of the traffic between member clusters. ([#4737](https://github.com/antrea-io/antrea/pull/4737) [#4606](https://github.com/antrea-io/antrea/pull/4606), [@hjiajing])

> - Add WireGuard encryption tunnel mode to Multi-cluster user guide. ([#4848](https://github.com/antrea-io/antrea/pull/4848), [@hjiajing])

Author: Updated.

- Add support for EndpointSlice API for Multi-cluster Services. When the EndpointSlice API is available for the cluster, EndpointSlice resources of the exported Service, rather than the Endpoints resource, will be watched. For the Pod IP type of Services, EndpointSlice resources will be exported as the Multi-cluster Service's endpoints. ([#4895](https://github.com/antrea-io/antrea/pull/4895), [@luolanzone])

> - Add IP based group association query API and ClientSet for querying the Antrea Group/ClusterGroups. ([#4807](https://github.com/antrea-io/antrea/pull/4807), [@Dyanngg])

Reviewer: "Add Antrea Controller API for querying Antrea Groups and ClusterGroups by IP addresses."

> - Add IP based group association query API and ClientSet for querying the Antrea Group/ClusterGroups. ([#4807](https://github.com/antrea-io/antrea/pull/4807), [@Dyanngg])

Author: Moved

> - Add new exporter to FlowAggregator to write flows to local file. ([#4855](https://github.com/antrea-io/antrea/pull/4855), [@antoninbas])

Reviewer: "a new exporter"
"a local file"

> - Support new OS openEuler 22.03 and add it in tested distributions. ([#4957](https://github.com/antrea-io/antrea/pull/4957), [@ceclinux])

Reviewer: "Add openEuler 22.03 as a new supported OS of Antrea, and update the [Kubernetes installer document](../docs/kubernetes-installers.md) with the information."

Author: Updated.

### Changed

- Deprecate Antrea Octant Plugin. ([#4825](https://github.com/antrea-io/antrea/pull/4825), [@antoninbas])
- Update Open vSwitch version to 2.17.6. ([#4959](https://github.com/antrea-io/antrea/pull/4959), [@tnqn])
- Update Windows OVS version to 2.16.7. ([#4705](https://github.com/antrea-io/antrea/pull/4705), [@XinShuYang])
- Upgrade K8s libraries to v0.26.4. ([#4935](https://github.com/antrea-io/antrea/pull/4935), [@heanlan])

> - Bump up whereabouts to v0.6.1. ([#4988](https://github.com/antrea-io/antrea/pull/4988), [@hjiajing])

Author: Done

> - Update Egress API to support multiple Egress IPs and pools. ([#4603](https://github.com/antrea-io/antrea/pull/4603), [@tnqn])

Reviewer 4: This is not really supported yet and might confuse users. It could say "Add `status.egressIP`  field for Egress to represent the effective Egress IP"

- Check the existence of AntreaAgentInfo CRD before operating on it for worker Node or ExternalNode. ([#4762](https://github.com/antrea-io/antrea/pull/4762), [@wenyingd])
- Stop serving v1alpha2 version of the ClusterGroup CRD. ([#4812](https://github.com/antrea-io/antrea/pull/4812), [@antoninbas])

> - Change to cache flow modification messages in Antrea Agent to improve Agent memory usage. ([#4495](https://github.com/antrea-io/antrea/pull/4495), [@wenyingd])

Reviewer 4: ```suggestion
- Optimize the cached flows in Antrea Agent to reduce Agent memory usage. ([#4495](https://github.com/antrea-io/antrea/pull/4495), [@wenyingd])
```

- Replace PacketIn/Controller with PacketIn2/Controller2 to improve packetin handler. ([#4768](https://github.com/antrea-io/antrea/pull/4768), [@GraysonWu])
- Change to look up Pods by name instead of IP address to fetch labels in Flow Aggregator, to avoid obtaining incorrect Pods when Pods turn over quickly. ([#4942](https://github.com/antrea-io/antrea/pull/4942), [@dreamtalen])
- Do not export Services of ExternalName type which is unsupported in upstream Multi-cluster Service. ([#4814](https://github.com/antrea-io/antrea/pull/4814), [@luolanzone])
- Update Multi-cluster user guide to provide more details for Gateway enablement. ([#4889](https://github.com/antrea-io/antrea/pull/4889), [@luolanzone])
- Update documentation for recent MetalLB versions. ([#4803](https://github.com/antrea-io/antrea/pull/4803), [@antoninbas])
- Add support for short-circuiting in AntreaProxy to ensure that the traffic from Pod/Node clients to
external addresses behaves the same way as the traffic from external clients to external addresses. ([#4815](https://github.com/antrea-io/antrea/pull/4815), [@hongliangl])

### Fixed

> - Use LOCAL instead of CONTROLLER as the in_port of packet-out message to fix Windows agent crash issue. ([#4992](https://github.com/antrea-io/antrea/pull/4992), [@tnqn])

Reviewer: message -> messages

a Windows agent crash issue

> - Fix the Antrea Agent crash issue on Windows by running modules that rely on Services after AntreaProxy is ready. ([#4946](https://github.com/antrea-io/antrea/pull/4946), [@tnqn])

Reviewer: To be consistent with the previous one, maybe change to "Run agent modules that rely on Services access after AntreaProxy is ready to fix a Windows agent crash issue"

- Fix Service not being updated correctly when stickyMaxAgeSeconds or InternalTrafficPolicy is updated. ([#4845](https://github.com/antrea-io/antrea/pull/4845), [@tnqn])
- Fix EndpointSlice API availablility check to resolve the issue that AntreaProxy always falls back to the Endpoints API when EndpointSlice is enabled. ([#4852](https://github.com/antrea-io/antrea/pull/4852), [@tnqn])
- In Antrea Agent Service CIDR discovery, prevent headless Services from updating the discovered Service CIDR to avoid overwriting the default route of host network unexpectedly. ([#5008](https://github.com/antrea-io/antrea/pull/5008), [@hongliangl])
- Use a random port when the UDP source port in a Traceflow is 0. ([#4963](https://github.com/antrea-io/antrea/pull/4963), [@gran-vmv])
- Set default flag to 2 for TCP Traceflow to fix a Traceflow timeout issue when the flag is not provided. ([#4948](https://github.com/antrea-io/antrea/pull/4948), [@luolanzone])
- Fix concurrent map write bug for LabelIdentity controller in Antrea Multi-cluster. ([#4994](https://github.com/antrea-io/antrea/pull/4994), [@Dyanngg])

> - Fix race condition between stale controller and ResourceImport reconcilers in Antrea Multi-cluster. ([#4853](https://github.com/antrea-io/antrea/pull/4853), [@Dyanngg])

Reviewer: a race condition

Reviewer: Antrea Multi-cluster Controller

> - Revert the change "Temporarily install Suricata from Ubuntu PPA instead of OISF PPA" to fix L7 NetworkPolicy failure. ([#4968](https://github.com/antrea-io/antrea/pull/4968), [@xliuxu])

Reviewer 4: ```suggestion
- Bump up Suricata to 6.0.12 to fix L7 NetworkPolicy issue. ([#4968](https://github.com/antrea-io/antrea/pull/4968), [@xliuxu])
```

Reviewer: a L7 NetworkPolicy failure

the L7 NetworkPolicy issue

> - Improvement in Windows cleanup scripts to avoid unexpected failure. ([#4722](https://github.com/antrea-io/antrea/pull/4722) [#5013](https://github.com/antrea-io/antrea/pull/5013), [@wenyingd])

Reviewer: Improve Windows cleanup scripts

failures

# Changelog 1.12

## 1.12.0 - 2023-05-24

> - The Multicast, TopologyAwareHints, NodeIPAM features are graduated from Alpha to Beta. The TopologyAwareHints, NodeIPAM features are enabled by default. Multicast can be enabled by a new configuration parameter `multicast.enable`.

Reviewer 2: > can be enabled with a new Antrea Agent configuration parameter: `multicast.enable`.

Author: Done

### Added

- Add two new fields `sourcePort` and `sourceEndPort` in Antrea-native policy API to match traffic initiated from specific ports. ([#4687](https://github.com/antrea-io/antrea/pull/4687), [@Dyanngg])

> - Add a new field `logLabel` to ACNP and ANP CRDs to specify firewall log label for Audit Logging. ([#4748](https://github.com/antrea-io/antrea/pull/4748), [@qiyueyao])

Reviewer 2: ```suggestion
- Add a new field `logLabel` to Antrea-native policy CRDs; the user-provided label is added to audit logs. ([#4748](https://github.com/antrea-io/antrea/pull/4748), [@qiyueyao])
```

Author: Updated.

- Update Open vSwitch version to 2.17.6. ([#4959](https://github.com/antrea-io/antrea/pull/4959), [@tnqn])
- Update Windows OVS version to 2.16.7. ([#4705](https://github.com/antrea-io/antrea/pull/4705), [@XinShuYang])
- Add `status.egressIP` field for Egress to represent the effective Egress IP. ([#4603](https://github.com/antrea-io/antrea/pull/4603), [@tnqn])
- Add a new `Failed` phase in ANP status for the case when all Agents have reported the status and at least one failure is received. ([#4608](https://github.com/antrea-io/antrea/pull/4608), [@wenyingd])
- Check the existence of AntreaAgentInfo CRD before operating on it for worker Node or ExternalNode. ([#4762](https://github.com/antrea-io/antrea/pull/4762), [@wenyingd])
- Stop serving v1alpha2 version of the ClusterGroup CRD. ([#4812](https://github.com/antrea-io/antrea/pull/4812), [@antoninbas])
- Optimize the cached flows in Antrea Agent to reduce Agent memory usage. ([#4495](https://github.com/antrea-io/antrea/pull/4495), [@wenyingd])
- Replace PacketIn/Controller with PacketIn2/Controller2 to improve packetin handler. ([#4768](https://github.com/antrea-io/antrea/pull/4768), [@GraysonWu])

> - Change to look up Pods by name instead of IP address to fetch labels in Flow Aggregator, to avoid obtaining incorrect Pods when Pods turn over quickly. ([#4942](https://github.com/antrea-io/antrea/pull/4942), [@dreamtalen])

Reviewer 2: > when Pod turnover is high

> - Do not export Services of ExternalName type which is unsupported in upstream Multi-cluster Service. ([#4814](https://github.com/antrea-io/antrea/pull/4814), [@luolanzone])

Reviewer 2: ```suggestion
- Do not export Services of type ExternalName in Antrea Multi-cluster; this is consistent with the upstream Multi-cluster Service KEP. ([#4814](https://github.com/antrea-io/antrea/pull/4814), [@luolanzone])
```

Author: Updated.
